doc/file_services.md

```diff
@@ -0,0 +1,745 @@
+reg      | general register, avoid 0, 1, 14 or 15.
+type     | INPUT, OUTPUT or UPDATE (default is INPUT).
+ioarea   | label of the I/O area for the record to be read/written.
+decbname | internal or external DECB for controlling READ/WRITE.
+rel      | relative record number.
+rba      | relative byte address.
+
+### DCB - Data Control Block
+
+Used to create the Data Control Block
+
+Some of the parameters may be set before and/or after OPEN.
+
+The individual descriptions indicate this.
+
+``` hlasm
+label    DCB   DSORG=PS,                                               X
+               EODAD=0,                                                X
+               RECFM=FB,                                               X
+               DDNAME=,                                                X
+               MACRF=GM,                                               X
+               SYNAD=0,                                                X
+               BLKSIZE=0,                                              X
+               LRECL=0,                                                X
+               DCBE=,                                                  X
+               RECORD=0,                                               X
+               DSNAME=0
+```
+
+EXLST is currently unsupported.
+
+'label' is the file identifier, the DCB name, which is used in all
+the file I/O macros.
+
+Term | Meaning                    | Description
+-----|----------------------------|------------
+BDW  | Block Descriptor Word      | Only used for RECFM=VB files. <br/>A 4-byte field of the form  'blocklength',H'0'<br/>The blocklength includes the length of the BDW.
+RDW  | Record Descriptor Word     |  Used for all variable files.<br/> A 4-byte field of the form H'recordlength',H'0'<br/>The recordlength includes the length of the RDW.<br/> The program must set this field correctly for output, and expect it on input.
+CRLF | Carriage Return, Line Feed | A term for the hex string X'0D0A' (ASCII) or X'0D25' (EBCDIC).<br/>Now commonly used to terminate a text record.
+
+
+#### DDNAME and DSNAME - DD statement/Dataset name
+
+DDNAME DSECT=DCBDDNAM Type=CL8 Default=CL8'label'
+
+DSNAME DSECT=DCBDSNAM Type=A No default
+
+* Can these be set in the DCB prior to OPEN: Yes
+* Can these be set in the DCB after OPEN : No
+
+Only one of these parameters may be set in the DCB.
+
+If both parameters are set in the program prior to open, the DSNAME will take precedence.
+
+When DDNAME is specified, an environment variable will point to the path and file that 
+is to be opened.
+
+Within a program the GETENV macro can be used to extract the environment
+variable. See the [Macro and Services Guide]() for full details.
+
+``` hlasm
+MYDCB    DCB   DDNAME=MYDATA, ...
+```
+
+For execution in a Windows environment:
+
+``` dos
+SET MYDATA=c:\path\file
+CALL ez390 c:\path\program parms
+```
+
+For execution in a \*nix environment
+
+``` sh
+MYDATA=/path/file ez390 program parms
```

Comment:
This item troubles me deeply.. 
Firstly, I think that the scripts/commands should work without providing the full path assuming that the script is in your PATH. The fact that you have to do this is strange and not normal. We should raise an issue to look at this as an enhancement.
Second, while I think this is useful information I am not sure it belongs here.
Once I start on the "User guide", I think this would be better in there. Until that exists, I will keep that info here. Changes will apply as per the suggestion

Reply:
Fixed 648dc18